### PR TITLE
Binomial Coefficient (under modulo) in dart

### LINKED ISSUE #5792: Binary Exponent calculation in Dart
## 💥 Proposal

implement an algo to calculate nCr under some mod in dart

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

yes

### PR DESCRIPTION
**Pull request by ashuvssut**
Binomial Coefficient (under modulo)  calculation in Dart

## Related Issues or Pull Requests

closes #5792

### CODE REVIEW

**Comment by ashuvssut** [comment]
@ankitaggarwal23 can you please ask for changes instead of _directly closing_ an open PR and an open issue😅just for a some small mistakes in README.md. I know maintainers are too busy. but atleast consider asking for changes.

writing a PR and issue again after getting closed doesnt feel right😓

Please pardon some mistakes and ask me for changes. I can do it right away

**Comment by ashuvssut** [comment]
@ankitaggarwal23 if the code changes given in this PR is not what NeoAlgo wants.. then i will close this PR myself. I understand if there is no need for the code I have written

But from your side kindly tell what kind of algorithm do you expect from contributors

contributors wont make such mistakes again

**Review by ankitaggarwal23 — COMMENTED** [review]
Don't push more than one code file in a single PR.

**Comment by ankitaggarwal23** [comment]
> Don't push more than one code file in a single PR.

Why you are updating and pushing other code file and readme.md

**Comment by ashuvssut** [comment]
> Don't push more than one code file in a single PR.

sir you can ask me to remove the code file. its the same code file that came through the recent merge I did.

why close this PR again😕

**Comment by ankitaggarwal23** [comment]
> > Don't push more than one code file in a single PR.
> 
> sir you can ask me to remove the code file. its the same code file that came through the recent merge I did.
> 
> why close this PR again😕

If you remove it will show three files in our side as well

**Comment by ankitaggarwal23** [comment]
You only see what you messed up with files

**Comment by ashuvssut** [comment]
No it wont show. I will do a squash commit

**Comment by ankitaggarwal23** [comment]
Add only your files and readme updates

**Comment by ashuvssut** [comment]
> You only see what you messed up with files

I rearranged the Readme alphabetically😅

**Comment by ashuvssut** [comment]
can you reopen this please

File: Dart/README.md [hunk 1]
```diff
@@ -17,22 +17,23 @@
 - [Armstrong number using Recursive approach](maths/ArmstrongRecursive.dart)
 - [Automorphic Number](maths/automorphic_number.dart)
 - [Binary Exponentiation (Iterative and Recursive methods)](maths/Binary_Exponentiation.dart)
-- [Catalan number (Recursion)](maths\catalan_number.dart)
+- [Binomial Coefficient (under modulo)](maths/binomial_coefficient_under_modulo.dart)
```
Comment (ashuvssut) [inline Dart/README.md]: this is the link to the related code file.

readme is arranged in alphabetical order

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Why you are pushing so many lines on readme. add you link only first you update your repo and then do the changes

**Comment by ashuvssut** [comment]
actually I arranged them in  alphabetical order

In the review I have marked the line that links the "Binary Exponent calculation" code

**Comment by ashuvssut** [comment]
ok i will remove the alphabetical order

**Comment by ashuvssut** [comment]
removed aphabetical order

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Can you confirm whether it is binary coefficient or binal exponent under modulo? Either make it same everywhere otherwise it becomes difficult for others to track

**Comment by ashuvssut** [comment]
in https://github.com/TesseractCoding/NeoAlgo/pull/5810 you changed the PR title to Binary Exponent calculation in Dart
So i also changed to that here. 

Now I have put the old one "Binomial Coefficient (under modulo)". Now there should be no confusion😄

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**